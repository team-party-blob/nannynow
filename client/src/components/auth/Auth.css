#auth {
  padding-top: 5vh;
  background-image: url('https://d2v9y0dukr6mq2.cloudfront.net/video/thumbnail/8vV2rpe/pastel-blue-background-1080p_x1kgy93b__F0000.png');
  background-size: cover;
  background-color: #eaf5ff;

  h1 {
    font-family: 'Noto Serif TC', serif;
    text-align: center;
    font-size: 40px;
  }

  #authForm {
    background-color: white;
    border: 3px solid grey;
    border-radius: 8px;
    display: flex;
    flex-direction: column;
    margin: 2em;
    padding: 1em;

    label {
      padding-top: 2em;
      padding-bottom: 0.5em;
      font-family: 'Noto Serif TC', serif;
      font-size: 18px;
    }

    input {
      height: 5vh;
      border-radius: 8px;
      border: 2px solid grey;

    }

    #role {
      text-align: center;
    }

    select {
      height: 5vh;
      border: 2px solid grey;
      font-family: 'Noto Serif TC', serif;
      font-size: 18px;
    }

    button {
      margin-top: 2em;
      margin-bottom: 2em;
      padding: 1em;
      border-radius: 8px;
      border: 2px solid grey;
      font-size: 18px;
      font-family: 'Noto Serif TC', serif;
      font-weight: 900;
      &:hover {
        background-color: #f96a54;
        cursor: pointer;
      }
    }
  }

  #switchUserType {
    text-align: center;

    h3 {
      font-family: 'Noto Serif TC', serif;
      font-size: 18px;
    }
    button {
      height: 5vh;
      padding-bottom: 2em;
      padding-top: 2em;
      margin-bottom: 2em;
      font-size: 18px;
      background-color: #f96a54;
      border: 2px solid grey;
      border-radius: 8px;
      font-family: 'Noto Serif TC', serif;
      font-weight: 900;
    }
  }
}

@media (min-width: 651px) {
  #auth {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    #authForm {
      align-self: center;
      width: 500px;
    }
  }
}
