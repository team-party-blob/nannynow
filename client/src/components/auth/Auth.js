import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ROUTES } from '../../routes/routes';

export default class Login extends PureComponent {

  static propTypes = {
    loginType: PropTypes.string.isRequired,
  };

  state = {
    email: '',
    password: '',
    userType: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  handleUserTypeChange = ({ target }) => {
    this.setState({ userType: target.value });
  };

  changeLoginType = () => {
    const { loginType } = this.props;
    if(loginType === 'Sign Up') this.props.history.push(ROUTES.SIGNIN.linkTo());
    if(loginType === 'Sign In') this.props.history.push(ROUTES.SIGNUP.linkTo());
  };

  handleSubmit = event => {
    event.preventDefault();
    const { email, password } = this.state;
    this.props.onSubmit({ email, password });
  };

  render() {

    const { email, password, userType } = this.state;
    const { loginType } = this.props;

    const nannyOrFamilyInput = () => {
      return (
        <div id='userType'>
          <label>Are You a Nanny or a Family?</label>
          <select name={userType} onChange={this.handleUserTypeChange}>
            <option value=''>Select</option>
            <option name={userType} key='nanny' value='nanny'>
              Nanny
            </option>
            <option name={userType} key='family' value='family'>
              Family
            </option>
          </select>
        </div>
      );
    };
    const alreadyUser = () => {
      return (
        <div>
          <h3>Already Have an Account?</h3>
          <button onClick={this.changeLoginType}>Click Here to Sign In</button>
        </div>
      );
    };

    const newUser = () => {
      return (
        <div>
          <h3>New to Nanny Now?</h3>
          <button onClick={this.changeLoginType}>Click Here to Sign Up</button>
        </div>
      );
    };

    return (
      <div>
        <h1>Welcome to Nanny Now!</h1>
        <form>
          <label htmlFor='email'>Email</label>
          <input
            type='email'
            name='email'
            value={email}
            onChange={this.handleChange}
          />
          <label>Password</label>
          <input
            type='password'
            name='password'
            value={password}
            onChange={this.handleChange}
          />
          {loginType === 'Sign Up' && nannyOrFamilyInput()}
          <button>{loginType}</button>
        </form>
        {loginType === 'Sign Up' && alreadyUser()}
        {loginType === 'Sign In' && newUser()}
      </div>
    );
  }
}
