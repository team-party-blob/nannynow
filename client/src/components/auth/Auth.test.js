import React from 'react';
import Auth from './Auth';
import { shallow } from 'enzyme';


jest.mock('../../routes/routes.js', () => ({
  ROUTES: {
    SIGNUP: {
      linkTo: () => '/'
    },
    SIGNIN: {
      linkTo: () => '/'
    }
  }
}));

describe('Auth component', () => {

  const onSubmit = jest.fn();
  
  it('matches snapshot', () => {
    const wrapper = shallow(<Auth loginType='Sign Up' onSubmit={onSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows appropriate questions for new user', () => {
    const wrapper = shallow(<Auth loginType='Sign Up' onSubmit={onSubmit} />);
    expect(wrapper.html()).toContain('Are You a Nanny or a Family?');
    expect(wrapper.html()).toContain('Already Have an Account?');
    expect(wrapper.html()).not.toContain('New to Nanny Now?');
  });

  it('shows appropriate questions for existing user', () => {
    const wrapper = shallow(<Auth loginType='Sign In' onSubmit={onSubmit} />);
    expect(wrapper.html()).not.toContain('Already Have an Account?');
    expect(wrapper.html()).not.toContain('Are You a Nanny or a Family?');
    expect(wrapper.html()).toContain('New to Nanny Now?');
  });
});
