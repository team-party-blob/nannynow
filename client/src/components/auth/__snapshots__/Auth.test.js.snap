// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth component matches snapshot 1`] = `
<div>
  <h1>
    Welcome to Nanny Now!
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      name="email"
      onChange={[Function]}
      type="email"
      value=""
    />
    <label>
      Password
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <div
      id="userType"
    >
      <label>
        Are You a Nanny or a Family?
      </label>
      <select
        name=""
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select
        </option>
        <option
          key="nanny"
          name=""
          value="nanny"
        >
          Nanny
        </option>
        <option
          key="family"
          name=""
          value="family"
        >
          Family
        </option>
      </select>
    </div>
    <button>
      Sign Up
    </button>
  </form>
  <div>
    <h3>
      Already Have an Account?
    </h3>
    <button
      onClick={[Function]}
    >
      Click Here to Sign In
    </button>
  </div>
</div>
`;
